plugins/FirmwareUpdateChecker/FirmwareUpdateChecker.py

```diff
@@ -21,32 +26,32 @@
 #  The plugin is currently only usable for applications maintained by Ultimaker. But it should be relatively easy
 #  to change it to work for other applications.
 class FirmwareUpdateChecker(Extension):
-    JEDI_VERSION_URL = "http://software.ultimaker.com/jedi/releases/latest.version?utm_source=cura&utm_medium=software&utm_campaign=resources"
 
-    def __init__(self):
+    def __init__(self) -> None:
         super().__init__()
 
-        # Initialize the Preference called `latest_checked_firmware` that stores the last version
-        # checked for the UM3. In the future if we need to check other printers' firmware
-        Application.getInstance().getPreferences().addPreference("info/latest_checked_firmware", "")
-
         # Listen to a Signal that indicates a change in the list of printers, just if the user has enabled the
-        # 'check for updates' option
+        # "check for updates" option
         Application.getInstance().getPreferences().addPreference("info/automatic_update_check", True)
         if Application.getInstance().getPreferences().getValue("info/automatic_update_check"):
             ContainerRegistry.getInstance().containerAdded.connect(self._onContainerAdded)
 
+        self._late_init = True  # Init some things after creation, since we need the path from the plugin-mgr.
```

Comment:
I did this as well (although it turned out that in that case, 'run' was called before pluginsLoaded was signalled). No matter, since the situation short-circuited itself as few commits later, and I could drop the late init.